contracts/UniswapV2Router01.sol

```diff
@@ -24,59 +20,39 @@ contract UniswapV2Router01 is IUniswapV2Router01, NilCurrencyBase {
     }
 
     function addLiquidity(
-        address tokenA,
-        address tokenB,
-        address to,
-        uint deadline,
-        uint salt
-    ) external override ensure(deadline) returns (uint amountA, uint amountB, uint liquidity) {
-        if (IUniswapV2Factory(factory).getTokenPair(tokenA, tokenB) == address(0)) {
-            IUniswapV2Factory(factory).createPair(tokenA, tokenB, 1, salt);
-        }
-        address pair = IUniswapV2Factory(factory).getTokenPair(tokenA, tokenB);
-        uint tokenAId = NilCurrencyBase(tokenA).getCurrencyId();
-        uint tokenBId = NilCurrencyBase(tokenB).getCurrencyId();
-
+        address pair,
+        address to
+    ) public override {
         Nil.Token[] memory tokens = Nil.msgTokens();
         if (tokens.length != 2) {
             revert("Send only 2 tokens to add liquidity");
         }
-        assert(tokenAId == tokens[0].id);
-        assert(tokenBId == tokens[1].id);
+        // TODO: Probably check that tokens are sent to the right pair
```

Comment:
Do we need this?

Reply:
Removed for now
But in the future it would be a good improvement to check that user sends right tokens to the right pair

Don't know how to implement this in case of pair is on another shard